### PR TITLE
Blog - Mastering Terraform Variables: A Comprehensive Guide for Defining and Utilizing Variable Types

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks

### CODE REVIEW

File: blog/_posts/2023-07-12-terraform-variables-guide.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+As an example, the following code shows a `region` variable of the type `string`, a `default` value of `us-west-2`, and a `description`:
+
+~~~{.tf caption=""}
+variable "region" {
+description = "The AWS region where resources will be created"
+type  = string
+default  = "us-west-2"
+}
+~~~
```
Comment (jalletto) [inline blog/_posts/2023-07-12-terraform-variables-guide.md:46]:
```suggestion
~~~{.tf caption=""}
variable "region" {
     description = "The AWS region where resources will be created"
     type  = string
     default  = "us-west-2"
}
~~~
```

Comment (jalletto) [reply blog/_posts/2023-07-12-terraform-variables-guide.md:46]: everything in the code blocks that's in {} should be indented.

File: blog/_posts/2023-07-12-terraform-variables-guide.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+
+1. **Reusability:** By utilizing variables in your IaaC configuration, you can avoid hardcoding values such as instance type or region names directly. This flexible approach allows you to utilize the same configuration files across multiple environments (*ie* development, testing, and production) or with different settings simply by changing the assigned values to the variable.
+2. **Security:** To minimize the risk of exposing sensitive data, it's recommended that you do not hardcode specific values, such as API keys and passwords, directly into your configurations. Instead, you should treat them as variables that can be dynamically provided when needed.
+
+You declare a Terraform variable using the `variable` block. The label after the `variable` keyword is the variable's name, which should be unique across all the variables in the same module:
+
+<div class="wide">
+![Terraform variable example]({{site.images}}{{page.slug}}/NgePUkR.png)
+</div>
+
```
Comment (jalletto) [inline blog/_posts/2023-07-12-terraform-variables-guide.md:35]: Can you just take the content of this image and turn it into a code block?

**Review by jalletto — CHANGES_REQUESTED** [review]

**Comment by msali123** [comment]
@jalletto Changes done

**Comment by msali123** [comment]
[Preview Link](https://649f0c780103a82a4a499330--ephemeral-narwhal-3ac3b2.netlify.app/blog/terraform-variables-guide/)

**Review by jalletto — APPROVED** [review]

**Pull request merged**